### DISCUSSION TITLE
UDP works slower than TCP

### DISCUSSION THREAD

**Discussion by UfkuAcik** [esnet/iperf#1503] [Q&A]
Hello everyone;

While I try to use,-u option with iperf3, usualy get around max 100 Mbit/s
At the same time if I don't use -u option I get 430 Mbit/s with TCP

It is documented that UDP supposed to be faster than TCP

Why such measurement is taking place?

**Comment by davidBar-On** [comment]
For comparison TCP and UDP, the same buffer (packet) length should be the same.  The default TCP length is 128KB while for UDP it is about 1.5KB.  Try setting both to the same value, e.g `-l 64K` (64K is the maximum UDP packet length), or at least increase the UDP packet length.

Another issue with UDP is that since it is not controlled, the client sending rate is limited only by CPU, which may cause the network buffers to overflow and therefore reduce throughput.  Therefore UDP throughput should be limited using the `-b` option and different values should be tried to get the maximum throughput.  E.g. try values between `-b 400M` and `-b 500M`.

One more option to increase UDP performance is the burst size - 30 may help.  E.g. `-b 450M/30`.
